vktrace/vktrace_layer/vktrace_lib_trim.cpp

```diff
@@ -1502,6 +1686,33 @@ void remove_Device_object(VkDevice var) {
         // If vktrace is in the middle of a trim, then we want to also generate calls to destroy any objects that have not yet been
         // destroyed by the app.
         add_destroy_device_object_packets(var);
+    } else {
+        // If vktrace is not in the middle of a trim, that mean it's in
+        // pre-trim, and the target app destroy the device before we start
+        // to trim (hot-key not pressed until now or it haven't reach
+        // the start-frame), the device object will not be used later, so
+        // we should not include related creation call for the device
+        // in final trimmed trace file. not only this device, any objects
+        // if created base on the device should be also excluded in final
+        // trimmed trace file. We should remove all these associate
+        // objects from trim track system to avoid generating any creation
+        // call for them.
+        //
+        // Commonly, if the target app destroy a device, it should already
+        // destroy all objects based on the device. But for some title,
+        // they directly destroy device without first destroy associate
+        // objects based on this device.
```

Comment:
It can be quite frustrating to try to trace/replay an app that seemingly works perfectly, but cannot not be traced/replayed because of incorrect usage of the API in the app. Trace/replay is a tool, and if the tool can work around problems such as this, it makes for a better experience for the trace/replay user. Yes, trace/replay can work around the bad API usage, but it should generate a warning.